driver/src/contracts/stablex_contract.rs

```diff
@@ -79,8 +78,27 @@ impl StableXContract for BatchExchange {
         //   request to run into the gas limit
         orders_builder.m.tx.gas = None;
         let packed_auction_bytes = orders_builder.call().wait()?;
-        let auction_data = parse_auction_data(packed_auction_bytes, index);
-        Ok(auction_data)
+        let mut reader = BatchedAuctionDataReader::new(index);
```

Comment:
I really like this abstraction you built there.

I am thinking it should live in the OrderbookReader directly instead of the contract. The contract was meant to be a thin layer providing an API for interacting with the smart contracts directly. Since we are using ethcontracts now, we might even get rid of it completely. I'd like for any logic (parsing, etc.) to live in a layer on top, that can be unit tested.

Can we change the contract API to return the raw bytes rather than a parsed result and do the parsing in the OrderBook?